client-server/client.cpp

```diff
@@ -0,0 +1,88 @@
+#include <stdio.h>
+#include "iostream"
+#include <stdlib.h>
+#include <unistd.h>
+#include <string>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/xml_parser.hpp>
+
+#define ServerPort	2100
+#define hostname	"victor-PC"
+#define filename	"info_settings_out.xml"
+#define MAXLEN		255
+struct messageToSend
+{
+	int len;
+	char buffer[256];
+	int from;
+	bool first;
+};
+
+using namespace std;
+
+int main(int argc, char *argv[])
+{	
+	using boost::property_tree::ptree;
+	int sockfd, portno, n;
+	struct sockaddr_in serv_addr;
+	struct hostent *server;
+	char buffer[256];
+    ptree tree;
+	stringstream sstream;
+	int length;
+	string toSend;
+	struct messageToSend m;
+	int index;
+
+	/*deschidere socket*/
+	sockfd = socket(AF_INET, SOCK_STREAM, 0);
+	if (sockfd < 0){
+		perror("ERROR opening socket");
+		exit(1);
+	}	
+	server = gethostbyname(hostname);
+	
+	if (server == NULL) {
+		fprintf(stderr,"ERROR, no such host\n");
+		exit(0);
+	}
+	bzero((char *) &serv_addr, sizeof(serv_addr));
+	serv_addr.sin_family = AF_INET;
+	
+	bcopy((char *)server->h_addr,
+		(char *)&serv_addr.sin_addr.s_addr,
+		server->h_length);
+
+	serv_addr.sin_port = htons(ServerPort);
+	if (connect(sockfd,(struct sockaddr *) &serv_addr,sizeof(serv_addr)) < 0) 
+		perror("ERROR connecting");
+
+	/*citire in stringstream-ul sstream tree-ul din xml*/
```

Comment:
If you have data stored as xml, and you want to send it as xml, why are you parsing it in a ptree rather than reading inside a buffer and sending it?